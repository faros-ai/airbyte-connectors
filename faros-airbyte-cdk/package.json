{
  "name": "faros-airbyte-cdk",
  "version": "0.0.1",
  "description": "Airbyte Connector Development Kit (CDK) for JavaScript/TypeScript",
  "keywords": [
    "airbyte",
    "cdk",
    "faros"
  ],
  "homepage": "https://www.faros.ai",
  "author": "Faros AI, Inc.",
  "license": "Apache-2.0",
  "repository": "https://github.com/faros-ai/airbyte-connectors",
  "bugs": {
    "url": "https://github.com/faros-ai/airbyte-connectors/issues"
  },
  "directories": {
    "lib": "lib",
    "test": "test"
  },
  "files": [
    "lib/",
    "resources/"
  ],
  "main": "./lib",
  "engines": {
    "node": ">=16"
  },
  "scripts": {
    "build": "tsc -p src",
    "clean": "rm -rf lib node_modules out",
    "fix": "prettier --write 'src/**/*.ts' 'test/**/*.ts' && npm run lint -- --fix",
    "lint": "eslint 'src/**/*.ts' 'test/**/*.ts'",
    "prepare": "npm run build",
    "test": "jest --verbose --color",
    "test-cov": "jest --coverage --verbose --color",
    "watch": "tsc -b -w src test"
  },
  "dependencies": {
    "@apidevtools/json-schema-ref-parser": "^9.0.9",
    "commander": "^9.3.0",
    "enquirer": "^2.3.6",
    "fast-redact": "^3.0.2",
    "json-schema-traverse": "^1.0.0",
    "lodash": "^4.17.21",
    "pino": "^8.5.0",
    "table": "^6.8.1",
    "verror": "^1.10.1"
  },
  "jest": {
    "coverageDirectory": "out/coverage",
    "coveragePathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "preset": "ts-jest",
    "testEnvironment": "node",
    "testTimeout": 10000,
    "globals": {
      "ts-jest": {
        "tsconfig": "test/tsconfig.json"
      }
    }
  }
}
