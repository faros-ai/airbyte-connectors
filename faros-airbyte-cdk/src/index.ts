export * from './destinations/destination';
export * from './destinations/destination-runner';
export {wrapApiError} from './errors';
export * from './help';
export * from './logger';
export * from './protocol';
export * from './sources/source-base';
export * from './sources/source';
export * from './sources/source-runner';
export * from './sources/streams/stream-base';
export * from './spec-loader';
export * from './utils';
