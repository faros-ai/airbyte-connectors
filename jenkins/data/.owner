2049311132 contextPath="" at 8@5674ab7e8d6d