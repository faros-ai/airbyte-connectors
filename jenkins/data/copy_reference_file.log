--- Copying files at Mon Sep  6 16:20:12 UTC 2021
--- Copying files at Mon Sep  6 19:51:19 UTC 2021
--- Copying files at Mon Sep  6 20:01:02 UTC 2021
