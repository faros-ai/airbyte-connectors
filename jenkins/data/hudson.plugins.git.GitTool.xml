<?xml version='1.1' encoding='UTF-8'?>
<hudson.plugins.git.GitTool_-DescriptorImpl plugin="git-client@3.9.0">
  <installations class="hudson.plugins.git.GitTool-array">
    <hudson.plugins.git.GitTool>
      <name>Default</name>
      <home>git</home>
      <properties/>
    </hudson.plugins.git.GitTool>
  </installations>
</hudson.plugins.git.GitTool_-DescriptorImpl>