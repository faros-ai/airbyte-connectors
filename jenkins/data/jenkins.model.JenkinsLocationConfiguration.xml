<?xml version='1.1' encoding='UTF-8'?>
<jenkins.model.JenkinsLocationConfiguration>
  <jenkinsUrl>http://localhost:8080/</jenkinsUrl>
</jenkins.model.JenkinsLocationConfiguration>