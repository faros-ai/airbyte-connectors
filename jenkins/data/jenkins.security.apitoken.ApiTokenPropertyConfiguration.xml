<?xml version='1.1' encoding='UTF-8'?>
<jenkins.security.apitoken.ApiTokenPropertyConfiguration>
  <tokenGenerationOnCreationEnabled>false</tokenGenerationOnCreationEnabled>
  <creationOfLegacyTokenEnabled>false</creationOfLegacyTokenEnabled>
  <usageStatisticsEnabled>true</usageStatisticsEnabled>
</jenkins.security.apitoken.ApiTokenPropertyConfiguration>