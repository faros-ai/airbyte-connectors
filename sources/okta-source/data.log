{"record":{"stream":"users","emitted_at":1641627744274,"data":{"id":"00u3jaazzxOxur0Bp5d7","status":"STAGED","created":"2022-01-08T06:50:58.000Z","activated":null,"statusChanged":null,"lastLogin":null,"lastUpdated":"2022-01-08T06:50:58.000Z","passwordChanged":null,"type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"yen","lastName":"hoang","mobilePhone":null,"secondEmail":null,"login":"yenhh@gmail.com","email":"yenhh@gmail.com"},"credentials":{"emails":[{"value":"yenhh@gmail.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3jaazzxOxur0Bp5d7"}},"groupsOfUser":[{"id":"00g3iw1gxoucJXLqk5d7","created":"2022-01-07T06:09:36.000Z","lastUpdated":"2022-01-07T06:09:36.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"BUILT_IN","profile":{"name":"Everyone","description":"All users in your organization"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/apps"}}},{"id":"00g3iwkxv69Dt4w3d5d7","created":"2022-01-07T07:53:52.000Z","lastUpdated":"2022-01-08T06:51:39.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"OKTA_GROUP","profile":{"name":"test group","description":"group description"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/apps"}}}]}},"type":"RECORD"}
{"record":{"stream":"users","emitted_at":1641627745523,"data":{"id":"00u3iw1h2t4oi09Cv5d7","status":"ACTIVE","created":"2022-01-07T06:09:40.000Z","activated":null,"statusChanged":"2022-01-07T07:43:31.000Z","lastLogin":"2022-01-08T07:32:08.000Z","lastUpdated":"2022-01-07T07:43:31.000Z","passwordChanged":"2022-01-07T07:43:31.000Z","type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"Kelvin","lastName":"Tran","mobilePhone":null,"secondEmail":null,"login":"kelvin.tr@mydevcube.com","email":"kelvin.tr@mydevcube.com"},"credentials":{"password":{},"emails":[{"value":"kelvin.tr@mydevcube.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3iw1h2t4oi09Cv5d7"}},"groupsOfUser":[{"id":"00g3iw1gxoucJXLqk5d7","created":"2022-01-07T06:09:36.000Z","lastUpdated":"2022-01-07T06:09:36.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"BUILT_IN","profile":{"name":"Everyone","description":"All users in your organization"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/apps"}}},{"id":"00g3iwkxv69Dt4w3d5d7","created":"2022-01-07T07:53:52.000Z","lastUpdated":"2022-01-08T06:51:39.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"OKTA_GROUP","profile":{"name":"test group","description":"group description"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/apps"}}}]}},"type":"RECORD"} 
{"record":{"stream":"groups","emitted_at":1641627748425,"data":{"id":"00g3iw1gxoucJXLqk5d7","created":"2022-01-07T06:09:36.000Z","lastUpdated":"2022-01-07T06:09:36.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"BUILT_IN","profile":{"name":"Everyone","description":"All users in your organization"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iw1gxoucJXLqk5d7/apps"}},"usersOfGroup":[{"id":"00u3iw1h2t4oi09Cv5d7","status":"ACTIVE","created":"2022-01-07T06:09:40.000Z","activated":null,"statusChanged":"2022-01-07T07:43:31.000Z","lastLogin":"2022-01-08T07:32:08.000Z","lastUpdated":"2022-01-07T07:43:31.000Z","passwordChanged":"2022-01-07T07:43:31.000Z","type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"Kelvin","lastName":"Tran","mobilePhone":null,"secondEmail":null,"login":"kelvin.tr@mydevcube.com","email":"kelvin.tr@mydevcube.com"},"credentials":{"password":{},"emails":[{"value":"kelvin.tr@mydevcube.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3iw1h2t4oi09Cv5d7"}}},{"id":"00u3jaazzxOxur0Bp5d7","status":"STAGED","created":"2022-01-08T06:50:58.000Z","activated":null,"statusChanged":null,"lastLogin":null,"lastUpdated":"2022-01-08T06:50:58.000Z","passwordChanged":null,"type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"yen","lastName":"hoang","mobilePhone":null,"secondEmail":null,"login":"yenhh@gmail.com","email":"yenhh@gmail.com"},"credentials":{"emails":[{"value":"yenhh@gmail.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3jaazzxOxur0Bp5d7"}}}]}},"type":"RECORD"}
{"record":{"stream":"groups","emitted_at":1641627749621,"data":{"id":"00g3iwkxv69Dt4w3d5d7","created":"2022-01-07T07:53:52.000Z","lastUpdated":"2022-01-08T06:51:39.000Z","lastMembershipUpdated":"2022-01-08T06:50:58.000Z","objectClass":["okta:user_group"],"type":"OKTA_GROUP","profile":{"name":"test group","description":"group description"},"_links":{"logo":[{"name":"medium","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-medium.1a5ebe44c4244fb796c235d86b47e3bb.png","type":"image/png"},{"name":"large","href":"https://ok12static.oktacdn.com/assets/img/logos/groups/odyssey/okta-large.d9cfbd8a00a4feac1aa5612ba02e99c0.png","type":"image/png"}],"users":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/users"},"apps":{"href":"https://dev-88164258.okta.com/api/v1/groups/00g3iwkxv69Dt4w3d5d7/apps"}},"usersOfGroup":[{"id":"00u3iw1h2t4oi09Cv5d7","status":"ACTIVE","created":"2022-01-07T06:09:40.000Z","activated":null,"statusChanged":"2022-01-07T07:43:31.000Z","lastLogin":"2022-01-08T07:32:08.000Z","lastUpdated":"2022-01-07T07:43:31.000Z","passwordChanged":"2022-01-07T07:43:31.000Z","type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"Kelvin","lastName":"Tran","mobilePhone":null,"secondEmail":null,"login":"kelvin.tr@mydevcube.com","email":"kelvin.tr@mydevcube.com"},"credentials":{"password":{},"emails":[{"value":"kelvin.tr@mydevcube.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3iw1h2t4oi09Cv5d7"}}},{"id":"00u3jaazzxOxur0Bp5d7","status":"STAGED","created":"2022-01-08T06:50:58.000Z","activated":null,"statusChanged":null,"lastLogin":null,"lastUpdated":"2022-01-08T06:50:58.000Z","passwordChanged":null,"type":{"id":"oty3iw1gxxYagRUHa5d7"},"profile":{"firstName":"yen","lastName":"hoang","mobilePhone":null,"secondEmail":null,"login":"yenhh@gmail.com","email":"yenhh@gmail.com"},"credentials":{"emails":[{"value":"yenhh@gmail.com","status":"VERIFIED","type":"PRIMARY"}],"provider":{"type":"OKTA","name":"OKTA"}},"_links":{"self":{"href":"https://dev-88164258.okta.com/api/v1/users/00u3jaazzxOxur0Bp5d7"}}}]}},"type":"RECORD"}
