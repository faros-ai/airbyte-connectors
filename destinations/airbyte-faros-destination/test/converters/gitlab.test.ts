import {AirbyteLog, AirbyteLogLevel} from 'faros-airbyte-cdk';
import _ from 'lodash';
import {getLocal} from 'mockttp';

import {CLI, read} from '../cli';
import {
  initMockttp,
  readTestResourceFile,
  tempConfig,
  testLogger,
} from '../testing-tools';
import {gitlabAllStreamsLog} from './data';

describe('gitlab', () => {
  const logger = testLogger();
  const mockttp = getLocal({debug: false, recordTraffic: false});
  const catalogPath = 'test/resources/gitlab/catalog.json';
  let configPath: string;
  const graphSchema = JSON.parse(readTestResourceFile('graph-schema.json'));
  const revisionId = 'test-revision-id';
  const streamNamePrefix = 'mytestsource__gitlab__';

  beforeEach(async () => {
    await initMockttp(mockttp);
    configPath = await tempConfig(mockttp.url);
  });

  afterEach(async () => {
    await mockttp.stop();
  });

  test('process and write records', async () => {
    await mockttp
      .forPost('/graphs/test-graph/models')
      .withQuery({schema: 'canonical'})
      .once()
      .thenReply(200, JSON.stringify({}));

    await mockttp
      .forPost('/graphs/test-graph/revisions')
      .once()
      .thenReply(
        200,
        JSON.stringify({
          entrySchema: graphSchema,
          revision: {uid: revisionId, lock: {state: {}}},
        })
      );

    let entriesSize = 0;
    await mockttp
      .forPost(`/graphs/test-graph/revisions/${revisionId}/entries`)
      .thenCallback(async (r) => {
        entriesSize = r.body.buffer.length;
        return {statusCode: 204};
      });

    await mockttp
      .forPatch(`/graphs/test-graph/revisions/${revisionId}`)
      .withJsonBodyIncluding({status: 'active'})
      .once()
      .thenReply(204);

    const cli = await CLI.runWith([
      'write',
      '--config',
      configPath,
      '--catalog',
      catalogPath,
    ]);
    cli.stdin.end(gitlabAllStreamsLog, 'utf8');

    const stdout = await read(cli.stdout);
    logger.debug(stdout);

    expect(stdout).toMatch('Wrote 77 records');
    expect(stdout).toMatch('Errored 0 records');
    expect(stdout).toMatch('Skipped 0 records');
    expect(await read(cli.stderr)).toBe('');
    expect(await cli.wait()).toBe(0);
    expect(entriesSize).toBeGreaterThan(0);
  });

  test('process records from all streams', async () => {
    const cli = await CLI.runWith([
      'write',
      '--config',
      configPath,
      '--catalog',
      catalogPath,
      '--dry-run',
    ]);
    cli.stdin.end(gitlabAllStreamsLog, 'utf8');

    const stdout = await read(cli.stdout);
    logger.debug(stdout);

    const processedByStream = {
      branches: 7,
      commits: 2,
      group_labels: 2,
      group_milestones: 1,
      groups: 1,
      issues: 2,
      jobs: 8,
      merge_request_commits: 6,
      merge_requests: 6,
      pipelines: 2,
      project_labels: 12,
      project_milestones: 2,
      projects: 1,
      releases: 1,
      tags: 1,
      users: 1,
    };
    const processed = _(processedByStream)
      .toPairs()
      .map((v) => [`${streamNamePrefix}${v[0]}`, v[1]])
      .orderBy(0, 'asc')
      .fromPairs()
      .value();

    const writtenByModel = {
      cicd_Build: 2,
      cicd_BuildCommitAssociation: 2,
      cicd_BuildStep: 8,
      cicd_Organization: 1,
      cicd_Pipeline: 1,
      cicd_Release: 1,
      cicd_ReleaseTagAssociation: 1,
      tms_Epic: 3,
      tms_Label: 15,
      tms_Project: 1,
      tms_Task: 2,
      tms_TaskAssignment: 1,
      tms_TaskBoard: 1,
      tms_TaskBoardProjectRelationship: 1,
      tms_TaskBoardRelationship: 2,
      tms_TaskProjectRelationship: 2,
      tms_TaskTag: 1,
      vcs_Branch: 7,
      vcs_BranchCommitAssociation: 7,
      vcs_Commit: 8,
      vcs_Organization: 1,
      vcs_PullRequest: 6,
      vcs_Repository: 1,
      vcs_Tag: 1,
      vcs_User: 1,
    };

    const processedTotal = _(processedByStream).values().sum();
    const writtenTotal = _(writtenByModel).values().sum();
    expect(stdout).toMatch(`Processed ${processedTotal} records`);
    expect(stdout).toMatch(`Would write ${writtenTotal} records`);
    expect(stdout).toMatch('Errored 0 records');
    expect(stdout).toMatch('Skipped 0 records');
    expect(stdout).toMatch(
      JSON.stringify(
        AirbyteLog.make(
          AirbyteLogLevel.INFO,
          `Processed records by stream: ${JSON.stringify(processed)}`
        )
      )
    );
    expect(stdout).toMatch(
      JSON.stringify(
        AirbyteLog.make(
          AirbyteLogLevel.INFO,
          `Would write records by model: ${JSON.stringify(writtenByModel)}`
        )
      )
    );
    expect(await read(cli.stderr)).toBe('');
    expect(await cli.wait()).toBe(0);
  });
});
