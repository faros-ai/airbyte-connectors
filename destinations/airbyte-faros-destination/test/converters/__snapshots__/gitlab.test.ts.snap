// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gitlab process records from all streams 1`] = `
Array [
  "Processed 55 records",
  "Processed records by stream: {\\\\\\"mytestsource__gitlab__branches\\\\\\":7,\\\\\\"mytestsource__gitlab__commits\\\\\\":2,\\\\\\"mytestsource__gitlab__group_labels\\\\\\":2,\\\\\\"mytestsource__gitlab__group_milestones\\\\\\":1,\\\\\\"mytestsource__gitlab__groups\\\\\\":1,\\\\\\"mytestsource__gitlab__issues\\\\\\":2,\\\\\\"mytestsource__gitlab__jobs\\\\\\":8,\\\\\\"mytestsource__gitlab__merge_request_commits\\\\\\":6,\\\\\\"mytestsource__gitlab__merge_requests\\\\\\":6,\\\\\\"mytestsource__gitlab__pipelines\\\\\\":2,\\\\\\"mytestsource__gitlab__project_labels\\\\\\":12,\\\\\\"mytestsource__gitlab__project_milestones\\\\\\":2,\\\\\\"mytestsource__gitlab__projects\\\\\\":1,\\\\\\"mytestsource__gitlab__releases\\\\\\":1,\\\\\\"mytestsource__gitlab__tags\\\\\\":1,\\\\\\"mytestsource__gitlab__users\\\\\\":1}\\"},\\"type\\":\\"LOG\\"}",
  "Would write 77 records",
  "Would write records by model: {\\\\\\"cicd_Build\\\\\\":2,\\\\\\"cicd_BuildCommitAssociation\\\\\\":2,\\\\\\"cicd_BuildStep\\\\\\":8,\\\\\\"cicd_Organization\\\\\\":1,\\\\\\"cicd_Pipeline\\\\\\":1,\\\\\\"cicd_Release\\\\\\":1,\\\\\\"cicd_ReleaseTagAssociation\\\\\\":1,\\\\\\"tms_Epic\\\\\\":3,\\\\\\"tms_Label\\\\\\":15,\\\\\\"tms_Project\\\\\\":1,\\\\\\"tms_Task\\\\\\":2,\\\\\\"tms_TaskAssignment\\\\\\":1,\\\\\\"tms_TaskBoard\\\\\\":1,\\\\\\"tms_TaskBoardProjectRelationship\\\\\\":1,\\\\\\"tms_TaskBoardRelationship\\\\\\":2,\\\\\\"tms_TaskProjectRelationship\\\\\\":2,\\\\\\"tms_TaskTag\\\\\\":1,\\\\\\"vcs_Branch\\\\\\":7,\\\\\\"vcs_BranchCommitAssociation\\\\\\":7,\\\\\\"vcs_Commit\\\\\\":8,\\\\\\"vcs_Organization\\\\\\":1,\\\\\\"vcs_PullRequest\\\\\\":6,\\\\\\"vcs_Repository\\\\\\":1,\\\\\\"vcs_Tag\\\\\\":1,\\\\\\"vcs_User\\\\\\":1}\\"},\\"type\\":\\"LOG\\"}",
  "Skipped 0 records",
  "Errored 0 records",
]
`;
